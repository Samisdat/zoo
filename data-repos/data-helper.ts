import path from "path";
import * as fs from "fs";

const rootDataDir = path.resolve(process.env.PWD as string, 'data-repos');

export type DataType = 'markdown' | 'geojson';

export const getDataDir = (dataType: DataType, dataSubDir:string): string =>{

    const dataDir = path.resolve(
        rootDataDir,
        dataType,
        dataSubDir
    );

    try {
        fs.lstatSync(dataDir).isDirectory()
    }
    catch (e){
        throw new Error(`dir [${dataDir}] does not exist`)
    }

    return dataDir;
};

export const getFileNames = async (dataType:DataType, dataSubDir:string): Promise<string[]> => {

    const dataDir = getDataDir(dataType, dataSubDir);

    let files = await fs.readdirSync(dataDir)

    files = files.filter((file)=>{

        if('.DS_Store' === file){
            return false
        }

        return true;

    });

    return files;

};