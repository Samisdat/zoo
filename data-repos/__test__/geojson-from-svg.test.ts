import {
    ensureGroupPresent,
    generateXmlContainingType,
    geojsonFromSvg, getElementIds, getGeojsonFromSvg,
    getSvg,
    getSvgPath,
    matchGroup, writeGeojsonContainingType, writeXmlContainingType
} from "../geojson-from-svg";
import {getDataDir} from "../data-helper";
import {forOfStatement} from "@babel/types";

describe('geojson from svg', () => {


    test('getSvgPath', async () => {

        expect(getSvgPath()).toBe('/Users/samisdat/repos/zoo/data-repos/svg/combined.svg');

    });

    test('getSvg', async () => {

        // ein sehr sehr guter test ;)
        expect(getSvg()).toContain('xml');

    });

    test('geojsonFromSvg exceptionThrown when type is not supported', ()=>{

        expect(
            ()=>{
                geojsonFromSvg('foobar')
            }
        ).toThrow(Error);

    });

    test('ensureGroupPresent', ()=>{

        const xml = '<g id="bounding-box">foobar</g>';

        expect(
            ensureGroupPresent(xml, 'bounding-box')
        ).toBeTruthy();

    });

    test('ensureGroupPresent', ()=>{

        const xml = '<g id="bounding-box">foobar</g>';

        expect(
            ensureGroupPresent(xml, 'facility-box')
        ).toBeFalsy();

    });

    test('matchGroup', ()=>{

        const xml = '<g id="bounding-box">foobar</g>';

        expect(
            matchGroup(xml, 'bounding-box')
        ).toMatchSnapshot();

    });

    test('generateXmlContainingType', ()=>{

        const xml = getSvg();

        expect(
            generateXmlContainingType(xml, 'bounding-box')
        ).toMatchSnapshot();

    });

    test.skip('writeXmlContainingType', ()=>{

        const xml = getSvg();

        expect(
            writeXmlContainingType(xml, 'bounding-box')
        ).toMatchSnapshot();

    });


    test('geojsonFromSvg', ()=>{

        expect(
            geojsonFromSvg('facility-circle')
        ).toMatchSnapshot();

    });

    test('getElementIds', ()=>{

        const xml = getSvg();

        const ids = getElementIds(xml);

        expect(ids).toMatchSnapshot();

    });

    test('getGeojsonFromSvg', async ()=>{

        const xml = getSvg();

        const dataSvg = generateXmlContainingType(xml, 'bounding-box')

        const geojson = await getGeojsonFromSvg(dataSvg, 'bounding-box');

        expect(geojson).toMatchSnapshot();

    });

    test.skip('writeGeojsonContainingType', async ()=>{

        const xml = getSvg();

        const dataSvg = generateXmlContainingType(xml, 'bounding-box')

        const geojson = await getGeojsonFromSvg(dataSvg, 'bounding-box');

        writeGeojsonContainingType(
            geojson,
            'bounding-box'
        );
    });

});



