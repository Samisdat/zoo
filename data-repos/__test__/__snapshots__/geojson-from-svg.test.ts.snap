// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`geojson from svg generateXmlContainingType 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?><!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\" \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<svg width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 2550 1994\\" version=\\"1.1\\"
    xmlns=\\"http://www.w3.org/2000/svg\\"
    xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xml:space=\\"preserve\\"
    xmlns:serif=\\"http://www.serif.com/\\" style=\\"fill-rule:evenodd;clip-rule:evenodd;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:1.5;\\">
    <MetaInfo xmlns=\\"http://www.prognoz.ru\\">
        <Geo>
            <GeoItem X=\\"0\\" Y=\\"0\\" Latitude=\\"51.24177020918754\\" Longitude=\\"7.105611562728882\\"/>
            <GeoItem X=\\"2550\\" Y=\\"1994\\" Latitude=\\"51.236776961813064\\" Longitude=\\"7.115809321403503\\"/>
        </Geo>
    </MetaInfo>
    <rect id=\\"bounding-box1\\" serif:id=\\"bounding-box\\" x=\\"4.997\\" y=\\"4.997\\" width=\\"2540.02\\" height=\\"1984.98\\" style=\\"fill:#fc0000;fill-opacity:0;\\"/>
</svg>"
`;

exports[`geojson from svg geojsonFromSvg 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?><!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\" \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<svg width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 2550 1994\\" version=\\"1.1\\"
    xmlns=\\"http://www.w3.org/2000/svg\\"
    xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xml:space=\\"preserve\\"
    xmlns:serif=\\"http://www.serif.com/\\" style=\\"fill-rule:evenodd;clip-rule:evenodd;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:1.5;\\">
    <MetaInfo xmlns=\\"http://www.prognoz.ru\\">
        <Geo>
            <GeoItem X=\\"0\\" Y=\\"0\\" Latitude=\\"51.24177020918754\\" Longitude=\\"7.105611562728882\\"/>
            <GeoItem X=\\"2550\\" Y=\\"1994\\" Latitude=\\"51.236776961813064\\" Longitude=\\"7.115809321403503\\"/>
        </Geo>
    </MetaInfo>
    <circle id=\\"Brillenpinguine2\\" serif:id=\\"Brillenpinguine\\" cx=\\"1734.55\\" cy=\\"296.25\\" r=\\"1.25\\"/><circle id=\\"Kirk-Dik-Dik\\" cx=\\"1459.55\\" cy=\\"78.649\\" r=\\"1.25\\"/><circle id=\\"Taskine2\\" serif:id=\\"Taskine\\" cx=\\"1724.55\\" cy=\\"1413.65\\" r=\\"1.25\\"/><circle id=\\"Rentiere1\\" serif:id=\\"Rentiere\\" cx=\\"1649.55\\" cy=\\"1468.65\\" r=\\"1.25\\"/><circle id=\\"Kiang1\\" serif:id=\\"Kiang\\" cx=\\"1529.55\\" cy=\\"1448.65\\" r=\\"1.25\\"/><circle id=\\"Juiniorzoo\\" cx=\\"2024.55\\" cy=\\"497.25\\" r=\\"1.25\\"/><circle id=\\"Löwenhaus1\\" serif:id=\\"Löwenhaus\\" cx=\\"1608.55\\" cy=\\"1796.75\\" r=\\"1.25\\"/><circle id=\\"Löwenhöhle\\" cx=\\"1938.55\\" cy=\\"1616.75\\" r=\\"1.25\\"/><circle id=\\"Löwenturm\\" cx=\\"2038.55\\" cy=\\"1571.75\\" r=\\"1.25\\"/><circle id=\\"Tiger3\\" serif:id=\\"Tiger\\" cx=\\"2263.55\\" cy=\\"1341.75\\" r=\\"1.25\\"/><circle id=\\"Okapis2\\" serif:id=\\"Okapis\\" cx=\\"1634.57\\" cy=\\"223.649\\" r=\\"1.25\\"/><circle id=\\"Südamerikahaus2\\" serif:id=\\"Südamerikahaus\\" cx=\\"1569.57\\" cy=\\"278.649\\" r=\\"1.25\\"/><circle id=\\"Vogelhaus2\\" serif:id=\\"Vogelhaus\\" cx=\\"1669.57\\" cy=\\"613.649\\" r=\\"1.25\\"/><circle id=\\"Freiflughalle1\\" serif:id=\\"Freiflughalle\\" cx=\\"1630.9\\" cy=\\"730.61\\" r=\\"1.25\\"/><circle id=\\"Yaks2\\" serif:id=\\"Yaks\\" cx=\\"1289.57\\" cy=\\"808.649\\" r=\\"1.25\\"/><circle id=\\"Okavango\\" cx=\\"1044.57\\" cy=\\"1078.65\\" r=\\"1.25\\"/><circle id=\\"LachenderHans\\" cx=\\"1119.57\\" cy=\\"1188.65\\" r=\\"1.25\\"/><circle id=\\"Rothunde2\\" serif:id=\\"Rothunde\\" cx=\\"1204.57\\" cy=\\"1143.65\\" r=\\"1.25\\"/><circle id=\\"Bären1\\" serif:id=\\"Bären\\" cx=\\"1344.57\\" cy=\\"948.649\\" r=\\"1.25\\"/><circle id=\\"Elefantenhaus1\\" serif:id=\\"Elefantenhaus\\" cx=\\"867.567\\" cy=\\"1302.15\\" r=\\"1.25\\"/><circle id=\\"Affenhaus2\\" serif:id=\\"Affenhaus\\" cx=\\"1039.57\\" cy=\\"873.649\\" r=\\"1.25\\"/><circle id=\\"KleinesAffenhaus\\" cx=\\"779.567\\" cy=\\"873.649\\" r=\\"1.25\\"/><circle id=\\"Känguros\\" cx=\\"514.567\\" cy=\\"908.649\\" r=\\"1.25\\"/><circle id=\\"Hirscheber2\\" serif:id=\\"Hirscheber\\" cx=\\"699.567\\" cy=\\"1150.65\\" r=\\"1.25\\"/><circle id=\\"Musikmuschel2\\" serif:id=\\"Musikmuschel\\" cx=\\"1043.07\\" cy=\\"533.649\\" r=\\"1.25\\"/><circle id=\\"Gibbons1\\" serif:id=\\"Gibbons\\" cx=\\"637.567\\" cy=\\"452.149\\" r=\\"1.25\\"/><circle id=\\"Eingang2\\" serif:id=\\"Eingang\\" cx=\\"861.438\\" cy=\\"309.839\\" r=\\"1.25\\" style=\\"fill-opacity:0.59;\\"/>
</svg>"
`;

exports[`geojson from svg getElementIds 1`] = `
Array [
  "Außengrenze",
  "Form der Wege",
  "Seelöwen",
  "Tiger",
  "Löwen",
  "Yaks",
  "Einsbären Haupt",
  "Eisbären hinten",
  "Patagonien",
  "Brillenpinguine",
  "Freigehege",
  "Elefantenbad",
  "Bullenanlage",
  "Elefantenhaus",
  "Elefanten",
  "Kiang",
  "Rentiere",
  "Taskine",
  "Kiosk Tigertal",
  "Katzenhaus",
  "Hirscheber",
  "Affenhaus",
  "Spielplatz",
  "Davidshirsche",
  "Kleines Affenhaus",
  "Okavagoo",
  "Tiger",
  "Seelöwen",
  "Brillenpinguine",
  "Südamerikahaus",
  "Biber",
  "Vogelhaus",
  "Gibbons",
  "Katzenhaus",
  "Löwen",
  "Lachender Hans",
  "Kangurus",
  "Okapis",
  "Kirk DikDiks",
  "JuniorZoo",
  "Steinböcke",
  "Pinguine",
  "Altes Ottergehege",
  "Eisbären",
  "Terrarium",
  "Aquarium",
  "Musikmuschel",
  "Eingang",
  "Yaks",
  "Ententeich",
  "Ziesel",
  "Tierwelt Patagoniens",
  "Emus",
  "Pelikane",
  "Bären",
  "Rothunde",
  "Zebras",
  "Waldrapp",
  "Pinselohrschweine",
  "Bongos",
  "Inseln",
  "Gibbonhaus",
  "Eingang",
  "Affenhaus",
  "Kleines Affenhaus",
  "KirkDikDik",
  "Zoosääle",
  "Südamerikahaus",
  "Okapis (wahrscheinlich)",
  "Okapis",
  "Gaukler",
  "Terraium",
  "Tigerkiosk",
  "Altes Löwenhaus",
  "Alte Direktion",
  "Pinguine",
  "Eisbären",
  "Waldrapp",
  "Zebras",
  "Pinselohrschweine",
  "Stall Steinböcke",
  "Stall Seehunde",
  "Zwergotter",
  "Beobachtung Bonobos",
  "Hütte Orange Utans",
  "Tunnel Gorilla",
  "Kleines Affenhaus",
  "Okawango",
  "Maras",
  "Stall Ganakos",
  "Ameisenbär",
  "Pelikane",
  "Emu",
  "Emu",
  "Lachender Hans",
  "Milus",
  "Chinesel",
  "Reentiere",
  "Taskine",
  "Biber",
  "Tiger",
  "Schneeleoparde",
  "Schneeleoparde",
  "Rote Pandas",
  "Katzenhaus",
  "Resterampe",
  "Ziegenstall",
  "Vogelhaus",
  "Freiflughalle",
  "Orange Utan Höhle",
  "Rothunde",
  "Bärenstall",
  "Kraniche",
  "Kanguroos",
  "Hirscheber",
  "Löwenhaus",
  "Waldrapp",
  "Musikmuschel",
  "WC Spielplatz",
  "Zooschule",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Kreuzung1",
  "2-3",
  "1-3",
  "1-2",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "3",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "3",
  "3",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "Alle Wege in einem Pfad",
  "3",
  "3",
  "5",
  "7",
  "6",
  "3-4",
  "4-5",
  "3-5",
  "5",
  "1",
  "1-3",
  "2-3",
  "1-2",
  "33",
  "5",
  "4",
  "3",
  "2",
  "1",
  "Brillenpinguine",
  "Kirk-Dik-Dik",
  "Taskine",
  "Rentiere",
  "Kiang",
  "Juiniorzoo",
  "Löwenhaus",
  "Löwenhöhle",
  "Löwenturm",
  "Tiger",
  "Okapis",
  "Südamerikahaus",
  "Vogelhaus",
  "Freiflughalle",
  "Yaks",
  "Okavango",
  "LachenderHans",
  "Rothunde",
  "Bären",
  "Elefantenhaus",
  "Affenhaus",
  "KleinesAffenhaus",
  "Känguros",
  "Hirscheber",
  "Musikmuschel",
  "Gibbons",
  "Eingang",
  "bounding-box",
]
`;

exports[`geojson from svg getGeojsonFromSvg 1`] = `
Object {
  "creator": "svg2geojson v0.7.1",
  "features": Array [
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            Array [
              7.105631546336763,
              51.24175769601948,
            ],
            Array [
              7.105631546336763,
              51.236787035952574,
            ],
            Array [
              7.1157893937833165,
              51.236787035952574,
            ],
            Array [
              7.1157893937833165,
              51.24175769601948,
            ],
            Array [
              7.105631546336763,
              51.24175769601948,
            ],
          ],
        ],
        "type": "Polygon",
      },
      "properties": Object {
        "name": "bounding-box",
        "type": "bounding-box",
      },
      "type": "Feature",
    },
  ],
  "type": "FeatureCollection",
}
`;

exports[`geojson from svg matchGroup 1`] = `
Array [
  "<g id=\\"bounding-box\\">foobar</g>",
  "",
  "foobar",
]
`;

exports[`geojson from svg writeXmlContainingType 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?><!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\" \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<svg width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 2550 1994\\" version=\\"1.1\\"
    xmlns=\\"http://www.w3.org/2000/svg\\"
    xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" xml:space=\\"preserve\\"
    xmlns:serif=\\"http://www.serif.com/\\" style=\\"fill-rule:evenodd;clip-rule:evenodd;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:1.5;\\">
    <MetaInfo xmlns=\\"http://www.prognoz.ru\\">
        <Geo>
            <GeoItem X=\\"0\\" Y=\\"0\\" Latitude=\\"51.24177020918754\\" Longitude=\\"7.105611562728882\\"/>
            <GeoItem X=\\"2550\\" Y=\\"1994\\" Latitude=\\"51.236776961813064\\" Longitude=\\"7.115809321403503\\"/>
        </Geo>
    </MetaInfo>
    <rect id=\\"bounding-box1\\" serif:id=\\"bounding-box\\" x=\\"4.997\\" y=\\"4.997\\" width=\\"2540.02\\" height=\\"1984.98\\" style=\\"fill:#fc0000;fill-opacity:0;\\"/>
</svg>"
`;
